internal/db/schema/migrations/oss/postgres/51/03_create_kms_key_destructions.up.sql

```diff
@@ -0,0 +1,69 @@
+begin;
+
+  create table kms_data_key_version_destruction_job (
+    key_id kms_private_id primary key -- one job per key version
+      references kms_data_key_version (private_id)
+        on delete cascade -- Note: instead of restrict, we cascade the delete to this job when the key version is deleted
+        on update cascade,
+    create_time wt_timestamp not null
+  );
+  comment on table kms_data_key_version_destruction_job is
+    'Table holding running and pending data key version destruction jobs';
+
+  create trigger immutable_columns before update on kms_data_key_version_destruction_job
+    for each row execute procedure immutable_columns('key_id', 'create_time');
+
+  create table kms_data_key_version_destruction_job_run (
+    key_id kms_private_id not null
+      references kms_data_key_version_destruction_job (key_id)
+        on delete cascade
+        on update cascade,
+    table_name text not null,
```

Comment:
I think this would add unnecessary toil when adding new encrypted material to the database. Today the application constructs any values in this column from the postgres reflection in https://github.com/hashicorp/go-kms-wrapping/blob/1c6d04de2431781864d990255855fde87c5e85ab/extras/kms/query.go#L4-L20.

Reply:
I wish I could make this a reference to that schema directly, but apparently foreign key references across schemas is not supported.

Reply:
Can we add a trigger that validates `table_name` is in the list returned by that query?

Reply:
Good idea!